lib/core/src/display/render/css3d/dom_container.rs

```diff
@@ -1,24 +1,33 @@
+// FIXME: ResizeCallback can be completely substituted by IntersectionCallback
+
 use crate::prelude::*;
 
 use crate::system::web::get_element_by_id;
 use crate::system::web::dyn_into;
 use crate::system::web::Result;
 use crate::system::web::StyleSetter;
 use crate::system::web::resize_observer::ResizeObserver;
+use crate::system::web::intersection_observer::IntersectionObserver;
 
 use wasm_bindgen::prelude::Closure;
 use web_sys::HtmlElement;
 use nalgebra::Vector2;
 use std::cell::RefCell;
 use std::rc::Rc;
 
+// ============================
+// === IntersectionCallback ===
+// ============================
+
+type IntersectionCallback        = Box<dyn Fn(&Vector2<f32>, &Vector2<f32>)>;
+pub trait IntersectionCallbackFn = Fn(&Vector2<f32>, &Vector2<f32>) + 'static;
```

Comment:
Lets add doc comment here. I did not know that something like Intersection Observer even exists. It would be nice to explain that in the code. Btw, what are we using it for ?

Reply:
I've added this comment:
// This file contains the implementation of DOMContainer. A struct that aids us to handle html
// elements, get its position and dimension avoiding style reflow.
//
// It relies on Resize Observer and Intersection Observer, which notifies us when the element's
// rect and visibility rect is updated.

Reply:
But this is not a module comment format! `//!` :)

Reply:
@wdanilo what is a module comment format? :z